src/include/cell_inline.h

```diff
@@ -268,6 +268,164 @@ __cell_pack_addr_validity(WT_SESSION_IMPL *session, uint8_t **pp, WT_TIME_AGGREG
+__wt_cell_kv_copy(WT_SESSION_IMPL *session, uint8_t *p, WT_CELL_KV *kv)
+{
+    size_t len;
+    uint8_t *t;
+
+    /*
+     * If there's only one chunk of data to copy (because the cell and data are being copied from
+     * the original disk page), the cell length won't be set, the WT_ITEM data/length will reference
+     * the data to be copied.
+     *
+     * WT_CELLs are typically small, 1 or 2 bytes -- don't call memcpy, do the copy in-line.
+     */
+    for (t = (uint8_t *)&kv->cell, len = kv->cell_len; len > 0; --len)
+        *p++ = *t++;
+
+    /* The data can be quite large -- call memcpy. */
+    if (kv->buf.size != 0)
+        memcpy(p, kv->buf.data, kv->buf.size);
+
+    WT_ASSERT(session, kv->len == kv->cell_len + kv->buf.size);
+}
+
+/*
+ * __wt_cell_pack_internal_key_addr --
+ *     Pack a key/value pair (internal page address or delta) directly into new_image.
+ */
+static WT_INLINE int
+__wt_cell_pack_internal_key_addr(WT_SESSION_IMPL *session, WT_ITEM *new_image,
+  WT_CELL_UNPACK_ADDR *base_key, WT_CELL_UNPACK_ADDR *base_val, WT_CELL_UNPACK_DELTA_INT *delta,
+  bool is_delta, uint8_t **pp)
+{
+    WT_CELL_KV key_kv, val_kv;
+    WT_PAGE_DELETED *page_del = NULL;
+    size_t packed_size;
+
+    WT_CLEAR(key_kv);
+    WT_CLEAR(val_kv);
+
+    /* Build packed key */
+    if (is_delta)
+        WT_RET(__cell_build_int_key_from_kv(session, &key_kv, delta->key.data, delta->key.size));
+    else
+        WT_RET(__cell_build_int_key_from_kv(session, &key_kv, base_key->data, base_key->size));
+
+    /* Build packed value */
+    if (is_delta) {
+        page_del = (delta->value.type == WT_CELL_ADDR_DEL) ? &delta->value.page_del : NULL;
+
+        __wt_cell_build_addr_kv(session, &val_kv, delta->value.type, page_del, &delta->value.ta,
+          delta->value.data, delta->value.size);
+
+    } else {
+        page_del = (base_val->type == WT_CELL_ADDR_DEL) ? &base_val->page_del : NULL;
+
+        __wt_cell_build_addr_kv(session, &val_kv, base_val->type, page_del, &base_val->ta,
+          base_val->data, base_val->size);
+    }
+
+    /*
+     * Ensure enough space, then recompute write pointer from new_image (not the caller's saved
+     * pointer)
+     */
+    packed_size = key_kv.len + val_kv.len;
+    if (new_image->size + packed_size > new_image->memsize)
+        WT_RET(__wt_buf_grow(session, new_image, new_image->size + packed_size));
+
+    /* Recompute write pointer after possible realloc */
+    WT_ASSERT(session, new_image->mem != NULL);
+
+    uint8_t *p = (uint8_t *)new_image->mem + new_image->size;
+    __wt_cell_kv_copy(session, p, &key_kv);
+    p += key_kv.len;
+    __wt_cell_kv_copy(session, p, &val_kv);
+    p += val_kv.len;
+
+    *pp = p;
+    new_image->size += packed_size;
+
+    __wt_buf_free(session, &key_kv.buf);
+    __wt_buf_free(session, &val_kv.buf);
+    return (0);
+}
+
+/*
+ * __wt_cell_build_addr_core --
+ *     Core function to pack an address cell. This version does not depend on reconcile-private
```

Comment:
No need to mention reconcile-private in the comment.

Reply:
__wt_cell_build_addr is enough?

Reply:
Renamed the function in 7f216e3f25678cf1ad74ac56e2e6cbe04b4d0094